crates/iceberg-rest-server/src/implementations/postgres/warehouse.rs

```diff
@@ -140,6 +154,237 @@ fn validate_warehouse_name(warehouse_name: &str) -> Result<()> {
+    })?;
+
+    Ok(())
+}
+
+pub(crate) async fn get_warehouse(
+    warehouse_id: &WarehouseIdent,
+    catalog_state: CatalogState,
+) -> Result<GetWarehouseResponse> {
+    let res = sqlx::query!(
+        r#"
+                SELECT 
+                    warehouse_id AS "warehouse_ident: WarehouseIdent",
+                    warehouse_name AS "warehouse_name: String",
+                    storage_profile AS "storage_profile: Value",
+                    storage_secret_id AS "storage_secret_id: Uuid",
+                    status AS "warehouse_status: WarehouseStatus"
+                FROM 
+                   warehouse
+                WHERE  
+                    warehouse_id = $1
+    
+        "#,
+        warehouse_id.as_uuid(),
+    )
+    .fetch_one(&catalog_state.read_pool)
+    .await
+    .map_err(|e| match e {
+        sqlx::Error::RowNotFound => warehouse_not_found(warehouse_id),
+        _ => e.into_error_model("Error fetching warehouses".to_string()),
+    })?;
+
+    Ok(GetWarehouseResponse {
+        warehouse_ident: res.warehouse_ident,
+        warehouse_name: res.warehouse_name,
+        storage_profile: deserialize_storage_profile(res.storage_profile)?,
+        warehouse_status: res.warehouse_status,
+        storage_credential_id: res.storage_secret_id.map(SecretIdent::from),
+    })
+}
+
+pub(crate) async fn deactivate_warehouse(
+    warehouse_id: &WarehouseIdent,
+    catalog_state: CatalogState,
+) -> Result<()> {
+    sqlx::query!(
+        r#"
+            UPDATE 
+                warehouse
+            SET 
+                status = 'inactive'
+            WHERE
+                warehouse_id = $1
+        "#,
+        warehouse_id.as_uuid(),
+    )
+    .execute(&catalog_state.write_pool)
+    .await
+    .map_err(|e| match e {
+        sqlx::Error::RowNotFound => warehouse_not_found(warehouse_id),
+        _ => e.into_error_model("Error fetching warehouses".to_string()),
+    })?;
+
+    Ok(())
+}
+
+pub(crate) async fn activate_warehouse(
+    warehouse_id: &WarehouseIdent,
+    catalog_state: CatalogState,
+) -> Result<()> {
+    sqlx::query!(
+        r#"
+            UPDATE 
+                warehouse
+            SET 
+                status = 'active'
+            WHERE
+                warehouse_id = $1
+        "#,
+        warehouse_id.as_uuid(),
+    )
+    .execute(&catalog_state.write_pool)
+    .await
+    .map_err(|e| match e {
+        sqlx::Error::RowNotFound => warehouse_not_found(warehouse_id),
+        _ => e.into_error_model("Error fetching warehouses".to_string()),
+    })?;
+
+    Ok(())
+}
+
+fn warehouse_not_found(warehouse_id: &WarehouseIdent) -> ErrorModel {
+    ErrorModel::builder()
+        .code(StatusCode::NOT_FOUND.into())
+        .message(format!("Warehouse not found: '{warehouse_id:?}'"))
+        .r#type("WarehouseNotFound")
+        .build()
+}
+
+fn serialize_storage_profile(storage_profile: StorageProfile) -> Result<Value, ErrorModel> {
```

Comment:
How do we do it in the "create_warehouse" function? Could we also use this function? Or make it part of the implementation block of StorageProfile?